File: experiments/source/ocaml/ratio_weighted_sum.ml
open Loom

let[@loom.entry] ratio_weighted_sum (scale : float) (epsilon : float) (x : Tensor1.t) (y : Tensor1.t) =
  Tensor1.reduce_sum
    (Tensor1.map2
       (fun xi yi -> (scale *. xi) /. (yi +. epsilon))
       x y)

File: experiments/source/python/ratio_weighted_sum.py
import loom


@loom.entry
def ratio_weighted_sum(scale: float, epsilon: float, x: loom.Tensor1, y: loom.Tensor1) -> float:
    return loom.Tensor1.reduce_sum(loom.Tensor1.map2(lambda xi, yi: (scale * xi) / (yi + epsilon), x, y))
